lib/Monads/NonDetMonad_Sat.thy

```diff
@@ -28,6 +28,29 @@ definition ex_exs_validE ::
   "\<lbrace>P\<rbrace> f \<exists>\<lbrace>Q\<rbrace>, \<lbrace>E\<rbrace> \<equiv> \<lbrace>P\<rbrace> f  \<exists>\<lbrace>\<lambda>rv. case rv of Inl e \<Rightarrow> E e | Inr v \<Rightarrow> Q v\<rbrace>"
 
 
+text \<open>
+  Seen as predicate transformer, this is the so-called conjugate wp in the literature, i.e. with
```

Comment:
I'm confused. What's seen as [a] predicate transformer?

Reply:
The definitions above the sentence (`exs_valid` and `exs_validE`). I can make that more explicit.

Reply:
Yes please

Reply:
Done!